leptos/src/children.rs

```diff
@@ -16,8 +16,62 @@ pub type ChildrenFnMut = Box<dyn FnMut() -> Fragment>;
 // This is to still support components that accept `Box<dyn Fn() -> Fragment>` as a children.
 type BoxedChildrenFn = Box<dyn Fn() -> Fragment>;
 
-#[doc(hidden)]
+/// This trait can be used when constructing a component that takes children without needing
+/// to know exactly what children type the component expects. This is used internally by the
+/// `view!` macro implementation, and can also be used explicitly when using the builder syntax.
+///
+/// # Examples
+///
+/// ## Without ToChildren
+///
+/// Without [ToChildren], consumers need to explicitly provide children using the type expected
+/// by the component. For example, [Provider][crate::Provider]'s children need to wrapped in
+/// a [Box], while [Show][crate::Show]'s children need to be wrapped in an [Rc].
+///
+/// ```
+/// # use leptos::{ProviderProps, ShowProps};
+/// # use leptos_dom::html::p;
+/// # use leptos_dom::IntoView;
+/// # use leptos_macro::component;
+/// # use std::rc::Rc;
+/// #
+/// #[component]
+/// fn App() -> impl IntoView {
+///     (
+///         ProviderProps::builder()
+///             .children(Box::new(|| p().child("Foo").into())),
+///         ShowProps::builder().children(Rc::new(|| p().child("Foo").into())),
+///     )
+/// }
+/// ```
+///
+/// ## With ToChildren
+///
+/// With [ToChildren], consumers don't need to know exactly which type a component uses for
+/// its children. This also removes the need to call `into()` on items in the children closure.
```

Comment:
> This also removes the need to call `into()` on items in the children closure.

I thought this was true because I wasn't getting an error in my IDE inside this example block, but it seems this is not true, and I do get an error if I try this outside of an example block.